agents/ten_packages/extension/fashionai/src/extension.py

```diff
@@ -0,0 +1,240 @@
+        CHANNEL = ten_env.get_property_string("channel")
+        STREAM_ID = str(ten_env.get_property_int("stream_id"))
+        if len(TOKEN) > 0 :
+            APP_ID = TOKEN
+        logger.info(f"FASHION_AI on_start: app_id = {APP_ID}, token = {TOKEN}, channel = {CHANNEL}, stream_id = {STREAM_ID}")
+
+        new_loop = asyncio.new_event_loop()
+
+        self.threadWebsocket = threading.Thread(target=self.thread_target, args=(new_loop,ten_env,APP_ID ,CHANNEL,STREAM_ID))
+        self.threadWebsocket.start()
+
+        threading.Thread(target=self.async_polly_handler, args=[ten_env]).start()
+
+        ten_env.on_start_done()
+
+    def on_stop(self, ten_env: TenEnv) -> None:
+        logger.info("FASHION_AI on_stop")
+        self.stopped = True
+        self.queue.put(None)
+        self.flush()
+        self.thread.join()
+
+        # TODO: clean up resources
+
+        ten_env.on_stop_done()
+
+    def on_deinit(self, ten_env: TenEnv) -> None:
+        logger.info("FASHION_AI on_deinit")
+        ten_env.on_deinit_done()
+
+    def on_cmd(self, ten_env: TenEnv, cmd: Cmd) -> None:
+        cmd_name = cmd.get_name()
+        logger.info("FASHION_AI on_cmd name {}".format(cmd_name))
+
+        # TODO: process cmd
+        if cmd_name == "flush":
+            self.outdate_ts = datetime.now()
+            self.flush()
+            cmd_out = Cmd.create("flush")
+            ten_env.send_cmd(cmd_out, lambda ten, result: logger.info("send_cmd flush done"))
+        else:
+            logger.info("unknown cmd {}".format(cmd_name))
+
+        cmd_result = CmdResult.create(StatusCode.OK)
+        ten_env.return_result(cmd_result, cmd)
+
+    def on_data(self, ten_env: TenEnv, data: Data) -> None:
+        # TODO: process data
+        inputText = data.get_property_string("text")
+        if len(inputText) == 0:
+            logger.info("FASHION_AI ignore empty text")
+            return
+
+        logger.info("FASHION_AI on data %s", inputText)
+        self.queue.put((inputText, datetime.now()))
+        # asyncio.get_event_loop().run_until_complete(self.client.send_inputText(inputText))
+        logger.info("FASHION_AI send_inputText %s", inputText)
+
+        # self.queue.put((inputText, datetime.now()))
+
+        pass
+
+    def on_audio_frame(self, ten_env: TenEnv, audio_frame: AudioFrame) -> None:
+        # TODO: process pcm frame
+        pass
+
+    def on_video_frame(self, ten_env: TenEnv, video_frame: VideoFrame) -> None:
+        # TODO: process image frame
+        pass            
+
+    async def init_fashionai(self, ten_env: TenEnv, app_id, channel, stream_id):
+        await self.client.connect()
+        await self.client.stream_start(app_id, channel, stream_id)
+        # asyncio.create_task(self.client.heartbeat(10))
+        await self.client.render_start()
+        while True:
+            await self.async_polly_handler()
+
+    def thread_target(self, loop, ten_env: TenEnv, app_id, channel, stream_id):
```

Comment:
maybe not necessary to make it instance method, just make it a local func in on_start?